### PR TITLE
Localize transformers list and throw from type-based transformers

### PR DESCRIPTION
**Pull request by captainsafia**
Some follow-up feedback to https://github.com/dotnet/aspnetcore/pull/57253.

### CODE REVIEW

File: src/OpenApi/src/Transformers/TypeBasedOpenApiOperationTransformer.cs [hunk 1]
```diff
@@ -27,6 +27,9 @@ internal IOpenApiOperationTransformer InitializeTransformer(IServiceProvider ser
         return transformer;
     }
 
+    /// <remarks>
+    /// Throw because the activate instance is invoked by the <see cref="OpenApiDocumentService" />.
+    /// </remarks>
     public Task TransformAsync(OpenApiOperation operation, OpenApiOperationTransformerContext context, CancellationToken cancellationToken)
-        => Task.CompletedTask;
+        => throw new InvalidOperationException("This method should not be called. Only activiated instances of this transformer should be used.");
```
Comment (martincostello) [inline src/OpenApi/src/Transformers/TypeBasedOpenApiOperationTransformer.cs:34]:
```suggestion
        => throw new InvalidOperationException("This method should not be called. Only activated instances of this transformer should be used.");
```

File: src/OpenApi/src/Transformers/TypeBasedOpenApiSchemaTransformer.cs [hunk 2]
```diff
@@ -27,7 +27,9 @@ internal IOpenApiSchemaTransformer InitializeTransformer(IServiceProvider servic
         return transformer;
     }
 
-    // No-op because the activate instance is invoked by the OpenApiSchema service.
+    /// <remarks>
+    /// Throw because the activate instance is invoked by the <see cref="OpenApiSchemaService" />.
+    /// </remarks>
     public Task TransformAsync(OpenApiSchema schema, OpenApiSchemaTransformerContext context, CancellationToken cancellationToken)
-        => Task.CompletedTask;
+        => throw new InvalidOperationException("This method should not be called. Only activiated instances of this transformer should be used.");
```
Comment (martincostello) [inline src/OpenApi/src/Transformers/TypeBasedOpenApiSchemaTransformer.cs:34]:
```suggestion
        => throw new InvalidOperationException("This method should not be called. Only activated instances of this transformer should be used.");
```

**Review by martincostello — APPROVED** [review]

**Review by BrennanConroy — APPROVED** [review]
We can save another alloc if we drop down to arrays over `List<T>` since we know the size upfront.

**Comment by amcasey** [comment]
"Localized" in this context means "converted from fields into locals"?

**Comment by captainsafia** [comment]
> "Localized" in this context means "converted from fields into locals"?

Yes 😸

**Pull request merged**